app/models/calculation_result.rb

```diff
@@ -148,13 +157,21 @@ def client_capital_rows
   end
 
   def client_capital_subtotal_rows
-    rows = {
-      total_capital: monetise(api_response.dig(:result_summary, :capital, :total_capital)),
-      pensioner_capital_disregard: monetise(-api_response.dig(:result_summary, :capital, :pensioner_capital_disregard)),
-      smod_disregard: monetise(-api_response.dig(:result_summary, :capital, :subject_matter_of_dispute_disregard)),
-    }
-
-    if has_partner? && pensioner_disregard_applied?
+    rows = if has_partner?
+             {
+               total_capital: monetise(api_response.dig(:result_summary, :capital, :total_capital)),
+               smod_disregard: monetise(-api_response.dig(:result_summary, :capital, :subject_matter_of_dispute_disregard)),
+               pensioner_capital_disregard: monetise(-api_response.dig(:result_summary, :capital, :pensioner_disregard_applied)),
+             }
+           else
+             {
+               total_capital: monetise(api_response.dig(:result_summary, :capital, :total_capital)),
+               pensioner_capital_disregard: monetise(-api_response.dig(:result_summary, :capital, :pensioner_disregard_applied)),
+               smod_disregard: monetise(-api_response.dig(:result_summary, :capital, :subject_matter_of_dispute_disregard)),
+             }
+           end
+
+    if (has_partner? && pensioner_disregard_applied?) || (pensioner_disregard_applied? && !pensioner_disregard_value.positive?)
```

Comment:
As above, if we update `pensioner_disregard_applied` then the logic here gets a lot simpler. We only show the pensioner capital disregard row if (a) its value is positive and (b) there is no partner. So we exclude that row if the value is zero or there is a partner.